### ISSUE TITLE
Over zooming

### ISSUE THREAD

**Issue by rochoa** [CartoDB/carto-vl#673]
tl;dr version: we should be able to use different zoom levels from the source to render different zoom levels in the layer/visualization.

The most basic example being rendering zoom level 15 with data from tiles at zoom level 14. Obviously, the tiles at the source should have enough resolution/detail.

This requires changes in the Source API to be able to say what zoom levels are available. The API could support two mechanisms:
1. Basic: as mentioned before, specifying the max zoom level, from there the visualization will use tiles from that zoom level to render higher zoom levels.
1. Advance: being able to map a source zoom level to a visualisation zoom level. This allows to just use some zoom levels in the source. E.g., your source only has data at zooms 0, 8, and 16, you can map those source zoom levels to visualisation zoom levels: `visZoom => max(visZoom//8 * 8, 16)`.

The _Advance_ mechanism will be useful for implementing https://github.com/CartoDB/carto-vl/issues/17. This requires changes in the backend and the client requesting the tiles with the right resolution, though. This comes from an offline chat I had with @davidmanzanares.

Related issues:
 - https://github.com/CartoDB/carto-vl/issues/17
 - https://github.com/CartoDB/carto-vl/issues/475

**Comment by alasarr** [comment]
+1
